src/constants.ts

```diff
@@ -0,0 +1,89 @@
+// @ts-ignore
+import systemPropTypes from '@styled-system/prop-types'
+import {themeGet} from '@styled-system/theme-get'
+import PropTypes from 'prop-types'
+import React from 'react'
+import * as styledSystem from 'styled-system'
+import theme from './theme'
+
+interface StyleFn extends styledSystem.styleFn {
+  propTypes?: React.WeakValidationMap<any>
+}
+
+const {get: getKey, compose, system} = styledSystem
+
+export const get = (key: string) => themeGet(key, getKey(theme, key))
+
+// Common props
+
+export const COMMON: StyleFn = compose(styledSystem.space, styledSystem.color, styledSystem.display)
+
+export interface CommonProps extends styledSystem.ColorProps, styledSystem.SpaceProps, styledSystem.DisplayProps {}
+
+COMMON.propTypes = {
+  ...systemPropTypes.space,
+  ...systemPropTypes.color,
+}
+
+// Typography props
+
+const whiteSpace = system({
+  whiteSpace: {
+    property: 'whiteSpace',
+    // cssProperty: 'whiteSpace',
+  },
+})
+
+export const TYPOGRAPHY: StyleFn = compose(styledSystem.typography, whiteSpace)
+
+export interface TypographyProps extends styledSystem.TypographyProps {
+  whiteSpace?: 'normal' | 'nowrap' | 'pre' | 'pre-wrap' | 'pre-line'
+}
+
+TYPOGRAPHY.propTypes = {
+  ...systemPropTypes.typography,
+  whiteSpace: PropTypes.oneOf(['normal', 'nowrap', 'pre-wrap', 'pre', 'pre-line']),
+}
+
+// Border props
+
+export const BORDER: StyleFn = compose(styledSystem.border, styledSystem.shadow)
+
+export interface BorderProps extends styledSystem.BorderProps, styledSystem.ShadowProps {}
+
+BORDER.propTypes = {
+  ...systemPropTypes.border,
+  ...systemPropTypes.shadow,
+}
+
+// Layout props
+
+export const LAYOUT: StyleFn = styledSystem.layout
+
+export interface LayoutProps extends styledSystem.LayoutProps {}
+
+LAYOUT.propTypes = systemPropTypes.layout
+
+// Position props
+
+export const POSITION: StyleFn = styledSystem.position
+
+export interface PositionProps extends styledSystem.PositionProps {}
+
+POSITION.propTypes = systemPropTypes.position
+
+// Flex props
+
+export const FLEX: StyleFn = styledSystem.flexbox
+
+export interface FlexProps extends styledSystem.FlexboxProps {}
```

Comment:
I wonder if it's worth calling these interfaces that extend `systemSystem.*` something like
 `SystemFlexProps` so that they don't get mixed up with the `FlexProps` that we'll eventually have for the `<Flex />` component?

`Flex` and `Grid` are probably the only collisions, but it might be nice to go ahead and change all of these if we like this change just to keep things consistent.

Reply:
Personally I would like to remove/deprecate `<Flex>`, `<Grid>`, `<Relative>`, `<Position>`, and even `<BorderBox>` and roll them all into `<Box>`. It seems `FlexProps` were recently added to `<Box>`.

Reply:
Oh, good catch, @dmarcey! I'll rename those interfaces.

@T-Hugs Thanks for bringing that up! I think both @emplums and I are open to revisiting the system prop API for Primer React components.